Makefile

```diff
@@ -16,6 +16,15 @@ ARCH?=amd64
 # Get version from git.
 GIT_VERSION?=$(shell git describe --tags --dirty)
 
+DOCKER_GO_BUILD := mkdir -p .go-pkg-cache && \
```

Comment:
I stole this from felix's makefile. I think it would be useful to replace a lot of the redundancy here with this base call. Will do in follow-up PR